src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -1152,5 +1191,39 @@ IReadOnlyList<Type> IInvocationBinder.GetParameterTypes(string methodName)
                 throw new InvalidOperationException($"There are no callbacks registered for the method '{methodName}'");
             }
         }
+
+        private class PeriodicConnectionTimer : IDisposable
+        {
+            public TimeSpan Interval { get; set; }
+            private Timer _timer;
+            private readonly TimerCallback _onTimeout;
+            private readonly object _state;
+
+            public PeriodicConnectionTimer(TimeSpan interval, object state, TimerCallback onTimeout)
+            {
+                Interval = interval;
+                _onTimeout = onTimeout;
+                _state = state;
+            }
+
+            public void Start()
+            {
+                _timer = new Timer(
+                    _onTimeout,
+                    _state,
+                    dueTime: Interval,
```

Comment:
And we generally don't use named args unless for some reasons we're passing in literal values.